File: C++/valid-sudoku.cpp
class Solution {
private:
    bool is_valid(std::vector<std::vector<char>> &board, int i, int j) {
        int val = board[i][j];
        board[i][j] = '.';
        for(int k=0; k<9; ++k) {
            if(board[i][k]==val || board[k][j]==val || board[(i/3)*3+k/3][(j/3)*3+k%3]==val) {
                board[i][j] = val;
                return false;
            }
        }
        board[i][j] = val;
        return true;
    }
public:
    bool isValidSudoku(std::vector<std::vector<char>> &board) {
        for(int i=0; i<9; ++i) {
            for(int j=0; j<9; ++j) {
                if('1'<=board[i][j] && board[i][j]<='9' && !is_valid(board, i, j))
                    return false;
            }
        }
        return true;
    }
};

File: Python/valid-sudoku.py
class Solution:
    # @param {character[][]} board
    # @return {boolean}
    def isValidSudoku(self, board):
        def is_valid(i, j):
            val = board[i][j]
            board[i][j] = '.'
            for k in xrange(9):
                if(board[i][k]==val or board[k][j]==val or board[(i/3)*3+k/3][(j/3)*3+k%3]==val):
                    board[i][j] = val
                    return False
            board[i][j] = val
            return True
        
        for i in xrange(9):
            for j in xrange(9):
                if '1'<=board[i][j]<='9' and not is_valid(i, j):
                    return False
        return True
